src/authn/index.js

```diff
@@ -91,10 +102,12 @@ function setup(app) {
     STRATEGY_OAUTH2,
     new OAuth2Strategy(
       {
-        authorizationURL: `${COGNITO_BASE_URL}/oauth2/authorize`,
-        tokenURL: `${COGNITO_BASE_URL}/oauth2/token`,
-        clientID: COGNITO_CLIENT_ID,
+        authorizationURL: OAUTH2_AUTHORIZATION_URL,
+        tokenURL: OAUTH2_TOKEN_URL,
+        clientID: OAUTH2_CLIENT_ID,
         callbackURL: "/logged-in",
+        scope: OAUTH2_SCOPES,
+        clientSecret: OAUTH2_CLIENT_SECRET,
```

Comment:
My understanding is that `OAUTH2_CLIENT_SECRET` is necessary for AAD and not Cognito because the `NextstrainDotOrgServerInstance-testing` IAM user has access to the user pool. And this Passport call is able to handle both cases. Is that correct?

Reply:
The Passport call is able to handle both the presence and absence of the client secret. A client secret is an optional part of client configuration in the spec. Azure AD requires one for the type of auth flow we're using and Cognito does not, though it supports them. For nextstrain.org, we could change to using a client secret with Cognito (and probably should). For Nextstrain CLI, as a public client, we can't secure a client secret and so wouldn't use one (true with Azure AD too).

Reply:
Ok, I sort of get it now. [This](https://stackoverflow.com/a/47156625) StackExchange answer was also helpful. I think I'll need to do more reading on OIDC/OAuth2 to fully grasp these concepts.

Reply:
Ok. Also happy to walk thru it in person!